File: iosApp/iosApp/Screens/PlaceholderScreens.swift
import SwiftUI

/// Minimal stand-in for every not-yet-built destination (U18): correct
/// header, declared back target, and the right room/field background so all
/// 16 routes are reachable in all three themes. U19–U22 replace these bodies.
struct PlaceholderScreen: View {
    let title: String
    let route: Route
    let onNavigate: (Route) -> Void
    var isFieldScreen: Bool = false

    @Environment(\.zinqqColors) private var colors

    var body: some View {
        let background = isFieldScreen ? colors.field : colors.dark
        let tint = isFieldScreen ? colors.onField : colors.onDark
        let muted = isFieldScreen ? colors.onFieldMuted : colors.onDarkMuted
        VStack(spacing: 0) {
            ScreenHeader(
                title: title,
                onBack: route.backTo.map { target in { onNavigate(target) } },
                tint: tint
            )
            Spacer()
            Text("TODO U19-U22")
                .font(ZinqqFont.sans(14))
                .foregroundColor(muted)
            Spacer()
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(background.ignoresSafeArea())
    }
}

File: androidApp/src/main/kotlin/zinqq/app/screens/PlaceholderScreens.kt
package zinqq.app.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.sp
import zinqq.app.nav.ScreenHeader
import zinqq.app.theme.ZinqqTheme

/**
 * Minimal stand-in for every not-yet-built destination (U13): correct header,
 * declared back target, and the right room/field background so all 16 routes
 * are reachable in all three themes. U14–U17 replace these bodies.
 */
@Composable
fun PlaceholderScreen(
    title: String,
    onBack: (() -> Unit)?,
    isFieldScreen: Boolean = false,
) {
    val colors = ZinqqTheme.colors
    val background = if (isFieldScreen) colors.field else colors.dark
    val tint = if (isFieldScreen) colors.onField else colors.onDark
    val muted = if (isFieldScreen) colors.onFieldMuted else colors.onDarkMuted
    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(background),
    ) {
        ScreenHeader(title = title, onBack = onBack, tint = tint)
        Box(
            modifier = Modifier
                .fillMaxSize()
                .weight(1f),
            contentAlignment = Alignment.Center,
        ) {
            Text(
                text = "TODO U14-U17",
                color = muted,
                fontSize = 14.sp,
            )
        }
    }
}
